flask_monitoringdashboard/core/cache.py

```diff
@@ -0,0 +1,111 @@
+"""
+    Contains the in memory cache used to increase the FMD performance.
+"""
+import datetime
+from multiprocessing import Lock
+
+from flask_monitoringdashboard.core.rules import get_rules
+from flask_monitoringdashboard.database import session_scope
+from flask_monitoringdashboard.database.endpoint import get_last_requested, get_endpoints_hits, get_endpoint_averages, \
+    update_last_requested
+
+memory_cache = {}
+mutex = Lock()
+
+
+class EndpointInfo(object):
+    """
+    Info about an endpoint that is stored in the memory cache.
+    """
+    def __init__(self, last_requested=None, average_duration=None, hits=None):
+        self.last_requested = last_requested
+        self.average_duration = average_duration if average_duration else 0
+        self.hits = hits if hits else 0
+
+    def set_last_requested(self, last_requested):
+        with mutex:
+            self.last_requested = last_requested
+
+    def set_duration(self, duration):
+        with mutex:
```

Comment:
do you really need the mutex here? 
i don't see any race conditions possible between set_last_requested, set_duration, and get_duration. or?

Reply:
All requests made to one endpoint will update the same EndpointInfo object. If two of them come in at the same time, they'll try to update the same `hits`, `average_duration`, and `last_requested` variables, which might result in incorrect results.